experiment/simulation/js/main.js

```diff
@@ -1 +1,610 @@
+      return window.clearInterval(interval);
+    }
+    yPos -= 0.6;
+    line.setAttribute("y1", `${yPos}%`);
+  }, 1);
+
+  // Filling the Jr cell slit
+  let path = document.getElementById("fill-slit");
+  let final = 1;
+  let cur = 0;
+  while (true) {
+    if (cur > final) {
+      break;
+    }
+    cur += 0.01;
+    path.setAttribute("offset", cur);
+    await new Promise((resolve) => setTimeout(resolve, 2));
+  }
+
+  document.getElementById("jrcell").style.cursor = "pointer";
+};
+
+async function moveJrCell(x) {
+  let image = document.getElementById("jrcell");
+  image.setAttribute("opacity", "1");
+  let a1 = anime.timeline({
+    targets: "#jrcell",
+    duration: 800,
+    easing: "linear",
+  });
+  if (x === 0 && overallIteration === 6) {
+    a1.add({
+      duration: 1000,
+      translateY: -300,
+      scale: 0.25,
+    }).add({
+      opacity: 0,
+    });
+
+    document.getElementById("instruction").innerHTML =
+      "Click on Observe button to observe what is happening inside the desiccator and choose video speed according to your own liking.";
+
+    document.getElementById("observation").innerHTML =
+      "Click on Observe button to observe what is happening inside the desiccator and choose video speed according to your own liking.";
+  } else if (x === 1) {
+    let a1 = anime.timeline({
+      targets: "#jrcell",
+      duration: 800,
+      easing: "linear",
+    });
+    a1.add({
+      duration: 1000,
+      translateY: 0,
+      scale: 1,
+    });
+  }
+}
+
+let fillPipette = async () => {
+  const line = document.getElementById("half-grad2");
+  const yFinalPosition = 0;
+  let yPos = 100;
+  const interval = window.setInterval(() => {
+    if (yPos < yFinalPosition) {
+      line.setAttribute("y1", "0.1%");
+      return window.clearInterval(interval);
+    }
+    yPos -= 0.6;
+    line.setAttribute("y1", `${yPos}%`);
+  }, 1);
+};
+
+function pur() {
+  if (overallIteration === 1) {
+    changeMessage();
+    let image = document.getElementById("spoon1");
+    image.setAttribute("opacity", "1");
+    image.style.transform = "translate(200%, -5%);";
+    image.style.pointerEvents = "none";
+    let a1 = anime.timeline({
+      targets: "#spoon1",
+      duration: 800,
+      easing: "linear",
+    });
+    a1.add({
+      duration: 0,
+      translateX: "280%",
+      translateY: "-125%",
+    })
+      .add({
+        duration: 800,
+        translateY: "-55%",
+      })
+      .add({
+        translateY: "-125%",
+        update: function (anim) {
+          document.getElementById("spoon-mouth").style.fill = "#b83dba";
+          document.getElementById("spoon-mouth").style.opacity = "1";
+        },
+      })
+      .add({
+        duration: 800,
+        translateX: "500%",
+        translateY: "175%",
+      })
+      .add({
+        delay: "800",
+        rotateZ: "45",
+      })
+      .add({
+        update: function (anim) {
+          document.getElementById("spoon-mouth").style.fill = "#b83dba";
```

Comment:
Please take the colors out in a separate variable or file with mnemonic keys and hexcode as values. Please take a look at https://github.com/virtual-labs/exp-atterberg-limits-iiith/blob/main/experiment/simulation/js/data.json